File: solutions/excel-sheet-column-number.cpp
/*
 *Solution of problem 171
 *https://leetcode.com/problems/excel-sheet-column-number/
 */
class Solution {
public:
    int titleToNumber(string s) {
        int size = s.size(), ans = 0;
        for(int i=0, j=size - 1; i<size; i++, j--){
            ans = ans + (s[i] - 'A' + 1) * pow(26, j);
        }
        return ans;
    }
};

File: solutions/excel-sheet-column-number.py
"""
171. Excel Sheet Column Number
- https://leetcode.com/problems/excel-sheet-column-number/
- https://leetcode.com/explore/challenge/card/august-leetcoding-challenge/550/week-2-august-8th-august-14th/3419/
"""
class Solution:
    def titleToNumber(self, s: str) -> int:
        result = 0
        sLen = len(s)
        for i in range(len(s)):
            result += (26**(sLen-1)) * (ord(s[i]) - ord('A') + 1)
            sLen -= 1
        return(result)
